### PR TITLE
Improved services management

### PR DESCRIPTION
**Pull request by inercia**
Do not try to restart the kubelet if 1) there are no manifests and 2) the kubelet is not installed.
Be smarter in some other things for services...

### CODE REVIEW

**Comment by inercia** [comment]
Need some more testing... do not merge yet.

**Comment by inercia** [comment]
After some testing, this seems to be working fine...

File: provision/minion/provision-salt-minion.sh [hunk 1]
```diff
@@ -77,10 +79,14 @@ mkdir -p /etc/salt/minion.d
 cp -v $SALT_ROOT_SUBDIR/config/minion.d/*  /etc/salt/minion.d
 [ -z $SKIP_ROLE_ASSIGNMENTS ] && cp -v $SALT_ROOT_SUBDIR/grains /etc/salt/
 
-if does_service_exist "kubelet" ; then
+if ls $MANIFESTS_DIR/* 1> /dev/null 2>&1 ; then
```
Comment (ereslibre) [inline provision/minion/provision-salt-minion.sh]: `ls $MANIFESTS_DIR/* &> /dev/null` ?

File: provision/dashboard/provision-dashboard.sh [hunk 2]
```diff
@@ -161,20 +166,23 @@ if [ -z "$FINISH" ] ; then
     mkdir -p "$K8S_MANIFESTS"
     echo "KUBELET_ARGS=\"--config=$K8S_MANIFESTS\"" > /etc/kubernetes/kubelet
 
-    log "Fixing etcd config and restarting it"
+    log "Fixing etcd config"
     sed -i 's@#\?ETCD_LISTEN_PEER_URLS.*@ETCD_LISTEN_PEER_URLS=http://0.0.0.0:2380@' /etc/sysconfig/etcd
     sed -i 's@#\?ETCD_LISTEN_CLIENT_URLS.*@ETCD_LISTEN_CLIENT_URLS=http://0.0.0.0:2379@' /etc/sysconfig/etcd
     sed -i 's@#\?ETCD_ADVERTISE_CLIENT_URLS.*@ETCD_ADVERTISE_CLIENT_URLS=http://dashboard:2379@' /etc/sysconfig/etcd
```
Comment (ereslibre) [inline provision/dashboard/provision-dashboard.sh]: Just a small note here (not related with this PR), but at some point we need to fix this at package level.

Comment (inercia) [inline provision/dashboard/provision-dashboard.sh]: Yeah, we will have to do something, but maybe we will move to an etcd container before that...

File: provision/minion/provision-salt-minion.sh (hunk 1 above)
Comment (inercia) [inline provision/minion/provision-salt-minion.sh]: Changed.

**Pull request merged**